src/environmentd/src/http.rs

```diff
@@ -726,16 +746,31 @@ async fn init_ws(
                 let creds = Credentials::Token { token };
                 (creds, options)
             }
+            WebSocketAuth::OptionsOnly { options: _ } => {
+                anyhow::bail!("Missing auth information!");
+            }
+        };
+        (auth(frontegg, creds).await?, options)
+    } else if let Some(user) = existing_user {
+        // If we're not using frontegg and we have an existing user provided by an
+        // upstream middleware, just use the options field from the auth message
+        match ws_auth {
+            WebSocketAuth::OptionsOnly { options }
+            | WebSocketAuth::Basic {
+                user: _,
```

Comment:
When is this ever possible? If we never expect basic auth AND a passed user header, this should error.

Reply:
This case allows the test to use `util::auth_with_ws(&mut ws, options).unwrap();` which uses basic auth; so similar to the console case I opted to ignore the other fields and just pull the `options` field if the user header is provided to avoid major code changes there.